fdbserver/RestoreLoader.actor.cpp

```diff
@@ -445,6 +436,8 @@ void _parseSerializedMutation(std::map<LoadingParam, VersionedMutationsMap>::ite
 
 		BackupStringRefReader kReader(k, restore_corrupted_data());
 		uint64_t commitVersion = kReader.consume<uint64_t>(); // Consume little Endian data
+		// We have already filter the commit not in [beginVersion, endVersion) when we concatenate kv pair in log file
+		ASSERT_WE_THINK(commitVersion >= asset.beginVersion && commitVersion < asset.endVersion);
```

Comment:
Might be useful to refactor `commitVersion >= asset.beginVersion && commitVersion < asset.endVersion` into a function like `asset.isInVersionRange(commitVersion)`